plugins/rest/rest_handlers.go

```diff
@@ -504,16 +416,31 @@ func (plugin *Plugin) telemetryNodeCountHandler(formatter *render.Render) http.H
 	}
 }
 
-// indexHandler - used to get index page
-func (plugin *Plugin) indexHandler(formatter *render.Render) http.HandlerFunc {
-	r := render.New(render.Options{
-		Directory:  "templates",
-		Asset:      Asset,
-		AssetNames: AssetNames,
-	})
-	return func(w http.ResponseWriter, req *http.Request) {
-		plugin.Log.Debugf("%v - %s %q", req.RemoteAddr, req.Method, req.URL)
+func (plugin *Plugin) registerIndexHandler(key, method string, indexItems []indexItem) {
+	handlerFunc := func(formatter *render.Render) http.HandlerFunc {
+		return func(w http.ResponseWriter, req *http.Request) {
+			plugin.Lock()
+			defer plugin.Unlock()
+
+			r := render.New(render.Options{
```

Comment:
This should not be inside the handler itself. It was intentionally declared before. Now the render instance is created for each request, which is not necessary.